src/installer/sources.py

```diff
@@ -139,6 +139,7 @@ def dist_info_filenames(self):
         return [
             name[len(base) + 1 :]
             for name in self._zipfile.namelist()
+            if name[-1] != "/"
```

Comment:
I wonder if we need to account for an empty string here. (easiest solution would be `name[-1:] != "/"`)

Reply:
I don't think this should be necessary because it looks at entries inside of the ZIP file but I will add it for extra safety.